composer/trainer/trainer.py

```diff
@@ -106,41 +44,281 @@
 
 __all__ = ["Trainer"]
 
-
-def _unpack_evaluators(
-    eval_dataloader: Union[Iterable, DataSpec, Evaluator, Sequence[Evaluator]],
+# syntax to shorten the Scheduler type annoations
+Scheduler = Union[ComposerScheduler, PyTorchScheduler]
+
+
+def _raise_missing_argument_exception(arg_name: str):
+    raise ValueError((f"{arg_name} is a required argument and must be specified when constructing the "
+                      f"{Trainer.__name__} or when calling {Trainer.__name__}.{Trainer.fit.__name__}(). "
+                      f"To fix, please specify `arg_name` via {Trainer.__name__}({arg_name}=...) or "
+                      f"{Trainer.__name__}.{Trainer.fit.__name__}({arg_name}=...)."))
+
+
+def _scale_max_duration_by_ssr(
+    scale_schedule_ratio: float,
+    orig_max_duration: Optional[Time[int]],
+) -> Optional[Time[int]]:
+    if orig_max_duration is None:
+        return None
+    max_duration = cast(Time[int], orig_max_duration * scale_schedule_ratio)
+    log.info(f'max_duration changed from {orig_max_duration} to {max_duration}')
+    if max_duration.value == 0:
+        raise ValueError('Scale schedule has reduced the max_duration to 0. Set a higher ratio or use more epochs.')
+    return max_duration
+
+
+def _should_step_schedulers_every_batch(schedulers: Optional[Union[Scheduler, Sequence[Scheduler]]]):
```

Comment:
```suggestion
def _get_default_scheduler_frequency(schedulers: Optional[Union[Scheduler, Sequence[Scheduler]]]):
```

Reply:
(This is really minor as it is a private helper) but since this function returns a boolean, I feel like `_get_default_scheduler_frequency` is ambiguous -- e.g. what does it mean when frequency is "True"? Having trouble thinking of a name that is shorter while also conveying that the returned boolean value means to step every batch.

Reply:
That's a good point. I don't have a good alternative either, maybe make a comment in the code return statement
```
# True means to step schedulers every batch, otherwise every epoch
return not pytorch_schedulers
```

or be explicit:
```
return TimeUnit.EPOCH if pytorch_schedulers else TimeUnit.BATCH
```

and then in the code do `step_schedulers_every_batch = _get... == TimeUnit.BATCH`

The latter seems a bit verbose, but it does leave flexibility down the road if we want to make this bool into `Time` interval.

Reply:
I like returning a TimeUnit -- makes it much cleaner.

Reply:
Resolved